components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilderTest.java

```diff
@@ -0,0 +1,241 @@
+
+        mockStatic(OAuthServerConfiguration.class);
+        when(OAuthServerConfiguration.getInstance()).thenReturn(oAuthServerConfiguration);
+        when(oAuthServerConfiguration.getTimeStampSkewInSeconds()).thenReturn((long) 3);
+        mockStatic(OAuth2Util.class);
+        when(oAuthServerConfiguration.getOpenIDConnectIDTokenExpiration()).thenReturn(String.valueOf((int) 78383));
+        when(OAuth2Util.mapDigestAlgorithm(any(JWSAlgorithm.class))).thenReturn(String.valueOf(JWSAlgorithm.RS256));
+
+        when(request1.getAuthorizedUser()).thenReturn(authenticatedUser);
+        when(authenticatedUser.getAuthenticatedSubjectIdentifier()).thenReturn("Subject_Identifier");
+
+        applicationManagementService = mock(ApplicationManagementService.class);
+        serviceProvider = mock(ServiceProvider.class);
+
+        mockStatic(OAuth2ServiceComponentHolder.class);
+        when(OAuth2ServiceComponentHolder.getApplicationMgtService()).thenReturn(applicationManagementService);
+        when(applicationManagementService.getServiceProviderByClientId(anyString(), anyString(), anyString())).thenReturn(serviceProvider);
+        when(tokenReqDTO.getClientId()).thenReturn("Client name");
+
+        JWSAlgorithm jwsAlgorithm1 = new JWSAlgorithm("Algo3");
+        when(OAuth2Util.mapSignatureAlgorithmForJWSAlgorithm(anyString())).thenReturn(jwsAlgorithm1);
+        when(OAuth2Util.mapDigestAlgorithm(any(JWSAlgorithm.class))).thenReturn("SHA-256");
+        DefaultIDTokenBuilder defaultIDTokenBuilder = new DefaultIDTokenBuilder();
+
+        mockStatic(MessageDigest.class);
+        when(MessageDigest.getInstance(String.valueOf(JWSAlgorithm.RS256))).thenReturn(messageDigest);
+        when(oAuth2AccessTokenRespDTO.getAccessToken()).thenReturn("AccessToken");
+
+        mockStatic(IdentityConfigParser.class);
+        when(IdentityConfigParser.getInstance()).thenReturn(identityConfigParser);
+        when(oAuthServerConfiguration.getOpenIDConnectCustomClaimsCallbackHandler()).thenReturn(customClaimsCallbackHandler);
+        doNothing().when(customClaimsCallbackHandler).handleCustomClaims(jwtClaimsSet, request);
+
+        when(OAuth2Util.signJWT(any(JWTClaimsSet.class), any(JWSAlgorithm.class), anyString())).thenReturn(jwt);
+        DefaultIDTokenBuilder defaultIDTokenBuilder1 = new DefaultIDTokenBuilder();
+        assertNotEquals(defaultIDTokenBuilder1.buildIDToken(request1, oAuth2AccessTokenRespDTO), "TestPasssed");
+    }
+
+    @Test
+    public void testBuildIDToken1() throws Exception {
+        request = mock(OAuthAuthzReqMessageContext.class);
+        identityProvider = mock(IdentityProvider.class);
+        identityProviderManager = mock(IdentityProviderManager.class);
+        authenticatedUser = mock(AuthenticatedUser.class);
+        when(request.getAuthorizationReqDTO()).thenReturn(oAuth2AuthorizeReqDTO);
+        when(oAuth2AuthorizeReqDTO.getTenantDomain()).thenReturn("carbon.super");
```

Comment:
Use a constant

Reply:
Better to use `MultitenantConstants.SUPER_TENANT_DOMAIN_NAME`